### DISCUSSION TITLE
Technical Team Meeting Minutes - 2020-12-09

### DISCUSSION THREAD

**Discussion by jwildfire** [SafetyGraphics/safetyGraphics#456] [Meetings]
## Date And Time

Dec 9, 2020 
1PM-2PM EST

## Attendees

- Jeremy Wildfire         
- Xiao Ni
- Melvin Munsaka
- Zak Skrivanek
- Bo Wang
- Sri Vimuri
- Nilesh Patel
- Renan Escalante

### 1. Project Review

Quick updates on [active projects](https://github.com/SafetyGraphics/SafetyGraphics.github.io/blob/master/roadmap.md)

Decisions: 
- We will merge pat profile project in to safetyCharts
- AE Volcano plot on hold until Isaac has time (Feb?)
- Jeremy to add links to repos

### 2. GitHub Discussion Intro

Github [released](https://github.blog/2020-05-06-new-from-satellite-2020-github-codespaces-github-discussions-securing-code-in-private-repositories-and-more/#discussions) a new discussion feature that I think will be useful for shared project management. I've activated the feature in safetyGraphics and safetyCharts, and am happy to do so on other repos if requested. 

### 3. {renv} workflow (Xiao)

Team discussed how updating to {renv} might change our development workflow. {renv} seems to be a great tool to capture a static snapshot of all the dependencies with their exact version numbers as used, e.g. [#446 Implement {renv}](https://github.com/SafetyGraphics/safetyGraphics/issues/446#issuecomment-734494263). On the other hand CRAN R packages' dependencies seem to be captured in the DESCRIPTION file only, and CRAN submission recommends testing against latest version of dependencies on CRAN.

Also discussed [Hadley's guidance](https://r-pkgs.org/description.html#versioning) that packages in the description file should have a minimum version listed.  There is a reference [article on its use in package development](https://rstudio.github.io/renv/articles/packages.html). More research is needed. 

Will revisit this topic as we get closer to v2. 

### 4. Chart Creation in SafetyGraphics 2.0

Making good progress here. [Draft Vignette](https://github.com/SafetyGraphics/safetyGraphics/wiki/Chart-Configuration-Vignette-(v2)) is here, and I'm actively updating/cleaning up safetyCharts with some [initial examples](https://github.com/SafetyGraphics/safetyCharts/tree/dev/R) in the `R` folder

**Comment by jwildfire** [comment]
Just updated the minutes and made the changes in the roadmap. @xni7 feel free to add some more details regarding {renv} / dependency management in a comment or a new discussion thread if my note above doesn't capture the details.

**Reply by xni7** [reply]
Added some more details
